lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -165,37 +176,34 @@ def prepare_data(d, init):
 
     def apply_callback(data):
         """A callback to apply changes."""
-        orig_xlim = axes.get_xlim()
-        orig_ylim = axes.get_ylim()
+        orig_limits = {
+            name: getattr(axes, f"get_{name}lim")()
+            for name in axis_map
+        }
 
         general = data.pop(0)
         curves = data.pop(0) if has_curve else []
         mappables = data.pop(0) if has_sm else []
         if data:
             raise ValueError("Unexpected field")
 
-        # Set / General
-        (title, xmin, xmax, xlabel, xscale, ymin, ymax, ylabel, yscale,
-         generate_legend) = general
+        title = general.pop(0)
+        axes.set_title(title)
+        generate_legend = general.pop()
 
-        if axes.get_xscale() != xscale:
-            axes.set_xscale(xscale)
-        if axes.get_yscale() != yscale:
-            axes.set_yscale(yscale)
+        for i, (name, axis) in enumerate(axis_map.items()):
+            axis_min = general[4*i]
+            axis_max = general[4*i + 1]
+            axis_label = general[4*i + 2]
+            axis_scale = general[4*i + 3]
+            if getattr(axes, f"get_{name}scale")() != axis_scale:
+                getattr(axes, f"set_{name}scale")(axis_scale)
 
-        axes.set_title(title)
-        axes.set_xlim(xmin, xmax)
-        axes.set_xlabel(xlabel)
-        axes.set_ylim(ymin, ymax)
-        axes.set_ylabel(ylabel)
-
-        # Restore the unit data
-        axes.xaxis.converter = xconverter
-        axes.yaxis.converter = yconverter
-        axes.xaxis.set_units(xunits)
-        axes.yaxis.set_units(yunits)
-        axes.xaxis._update_axisinfo()
-        axes.yaxis._update_axisinfo()
+            getattr(axes, f"set_{name}lim")(axis_min, axis_max)
+            axis.set_label_text(axis_label)
+            axis.converter = axis_converter[name]
```

Comment:
Let's keep the original comment to make it clear that these commands belong together
```suggestion

            # Restore the unit data
            axis.converter = axis_converter[name]
```